pkg/configuration/parameters_test.go

```diff
@@ -82,18 +81,78 @@ func TestPostParse(t *testing.T) {
 		assert.True(t, param.Meta.Ready(), "method Ready() returned false")
 	})
 
+	t.Run("app parameters NOT VALID NOR READY from INVALID inputs (MISSING URL)", func(t *testing.T) {
+		// Arrange
+		param = convertToJiraApiResourceParameters(param)
+		*param.JiraAPIUrl = ""
+		context = "ReadIssue"
+		issueList = "ABC-123 DEF-456"
+
+		// Act
+		param.InitializeAndValidatePostParse(&context, &issueList)
+
+		// Assert
+		assert.False(t, param.Meta.AllMandatoryValuesPresent(), "method AllMandatoryValuesPresent() returned true")
+		assert.False(t, param.Meta.Ready(), "method Ready() returned true")
+	})
+
+	t.Run("app parameters NOT VALID NOR READY from INVALID inputs (MISSING USERNAME)", func(t *testing.T) {
+		// Arrange
+		param = convertToJiraApiResourceParameters(param)
+		*param.Username = ""
+		context = "ReadIssue"
+		issueList = "ABC-123 DEF-456"
+
+		// Act
+		param.InitializeAndValidatePostParse(&context, &issueList)
+
+		// Assert
+		assert.False(t, param.Meta.AllMandatoryValuesPresent(), "method AllMandatoryValuesPresent() returned true")
+		assert.False(t, param.Meta.Ready(), "method Ready() returned true")
+	})
+
+	t.Run("app parameters NOT VALID NOR READY from INVALID inputs (MISSING PASSWORD)", func(t *testing.T) {
+		// Arrange
+		param = convertToJiraApiResourceParameters(param)
+		*param.Password = ""
+		context = "ReadIssue"
+		issueList = "ABC-123 DEF-456"
+
+		// Act
+		param.InitializeAndValidatePostParse(&context, &issueList)
+
+		// Assert
+		assert.False(t, param.Meta.AllMandatoryValuesPresent(), "method AllMandatoryValuesPresent() returned true")
+		assert.False(t, param.Meta.Ready(), "method Ready() returned true")
+	})
+
+	t.Run("app parameters NOT VALID NOR READY from INVALID inputs (EMPTY ISSUES)", func(t *testing.T) {
+		// Arrange
+		param = convertToJiraApiResourceParameters(param)
+		param.IssueList = make([]string, 0)
+		context = "ReadIssue"
+		issueList = ""
+
+		// Act
+		param.InitializeAndValidatePostParse(&context, &issueList)
+
+		// Assert
+		assert.True(t, param.Meta.AllMandatoryValuesPresent(), "method AllMandatoryValuesPresent() returned false")
+		assert.False(t, param.Meta.Ready(), "method Ready() returned true")
+	})
+
 }
 
-func convertToJiraApiResourceParameters() configuration.JiraAPIResourceParameters {
-	param := configuration.JiraAPIResourceParameters{}
-	param.JiraAPIUrl = &tParam.JiraAPIUrl
-	param.Username = &tParam.Username
-	param.Password = &tParam.Password
-	param.Destination = &tParam.Destination
-	param.ClosedStatusName = &tParam.ClosedStatusName
-	param.TransitionName = &tParam.TransitionName
-
-	param.EditCustomFieldParam.CustomFieldName = &tParam.EditCustomFieldParam.CustomFieldName
+func convertToJiraApiResourceParameters(param configuration.JiraAPIResourceParameters) configuration.JiraAPIResourceParameters {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [func convertToJiraApiResourceParameters should be convertToJiraAPIResourceParameters](https://app.codacy.com/manual/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=5050785)